x-pack/plugins/integration_assistant/server/graphs/ecs/validate.ts

```diff
@@ -46,43 +45,97 @@ function findMissingFields(combinedSamples: string, ecsMapping: AnyObject): stri
+  if (typeof value !== 'object' || value === null || Array.isArray(value)) {
+    return null;
+  }
+
+  if (
+    value.target &&
+    typeof value.target === 'string' &&
+    value.type &&
+    typeof value.type === 'string'
+  ) {
+    let confidence = 0.5;
+    if (value.confidence && typeof value.confidence === 'number') {
+      confidence = value.confidence;
+    }
+    let dateFormats: string[] = [];
+    if (value.date_formats && Array.isArray(value.date_formats)) {
+      dateFormats = value.date_formats;
+    } else if (value.date_format && Array.isArray(value.date_format)) {
+      dateFormats = value.date_format;
+    } else if (value.date_format && typeof value.date_format === 'string') {
+      dateFormats = [value.date_format];
+    }
+    return {
+      target: value.target,
+      type: value.type,
+      confidence,
+      date_formats: dateFormats,
+    };
+  }
+
+  return null;
+}
+
+/**
+ * Extracts ECS (Elastic Common Schema) field mapping dictionary from the LLM output.
+ *
+ * @param path - The current path in the object being traversed (an array of strings).
+ * @param value - The value to be processed, which can be an array, object, or other types.
+ * @param output - A record where the extracted ECS mappings will be stored. The keys are ECS targets, and the values are arrays of paths.
+ *
+ * This function recursively traverses the provided value. If the value is an array, it processes each item in the array.
+ * If the value can be interpreted as an ECS mapping, it adds the path to the output record under the appropriate ECS target.
+ * If the value is a regular object, it continues traversing its properties.
+ */
+export function extractECSMapping(
   path: string[],
   value: any,
   output: Record<string, string[][]>
 ): void {
-  if (typeof value === 'object' && value !== null) {
-    if (!Array.isArray(value)) {
-      // If the value is a dict with all the keys returned for each source field, this is the full path of the field.
-      const valueKeys = new Set(Object.keys(value));
-
-      if ([...valueFieldKeys].every((k) => valueKeys.has(k))) {
-        if (value?.target !== null) {
-          if (!output[value?.target]) {
-            output[value.target] = [];
-          }
-          output[value.target].push(path);
-        }
-      } else {
-        // Regular dictionary, continue traversing
-        for (const [k, v] of Object.entries(value)) {
-          processMapping([...path, k], v, output);
-        }
-      }
-    } else {
-      // If the value is an array, iterate through items and process them
-      for (const item of value) {
-        if (typeof item === 'object' && item !== null) {
-          processMapping(path, item, output);
-        }
+  if (Array.isArray(value)) {
+    // If the value is an array, iterate through items and process them.
+    for (const item of value) {
+      if (typeof item === 'object' && item !== null) {
+        extractECSMapping(path, item, output);
```

Comment:
Can try something like. Just a suggestion feel free to pick whatever you like to

```TS
value
 .filter((item) => (typeof item === 'object' && item !== null))
 .forEach((item) => extractECSMapping(path, item, output));
```

Reply:
I agree it can be rewritten to be cleaner, but let's keep it like this because this makes it clear I just moved the lines around.